src/framework/mod.rs

```diff
@@ -48,6 +50,10 @@ pub struct HttpApiClient {
     http_client: reqwest::Client,
 }
 
+pub struct HttpApiClientConfig {
```

Comment:
I would say instead of making every field an `Option`, perhaps we should use https://doc.rust-lang.org/std/default/trait.Default.html